src/sas/qtgui/Perspectives/Invariant/InvariantPerspective.py

```diff
@@ -1,21 +1,25 @@
 # global
 import copy
 import logging
+import math
 
 import numpy as np
 from PySide6 import QtCore, QtGui, QtWidgets
 from twisted.internet import reactor, threads
 
 import sas.qtgui.Utilities.GuiUtils as GuiUtils
+from sas.qtgui.Perspectives.Corfunc.CorfuncSlider import CorfuncSlider
+from sas.qtgui.Plotting import PlotterData
 from sas.qtgui.Plotting.PlotterData import Data1D, DataRole
+from sas.sascalc.corfunc.calculation_data import ExtrapolationInteractionState, ExtrapolationParameters
```

Comment:
We should look to extract these common routines from the `Corfunc` perspective so that we do not introduce cross-perspective dependencies.